### PR TITLE
feat(cli): add cli sort flag to subcommands

### LINKED ISSUE #1625: [Feat] zli: support command line options for sorting criteria
Right now all results are sorted alphabetically, we should support more options.
This issue depends on the fix for https://github.com/project-zot/zot/issues/1624

### PR DESCRIPTION
**Pull request by laurentiuNiculae**
**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:
Closes #1625

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/cli/image_cmd_test.go [hunk 1]
```diff
@@ -1356,6 +1356,85 @@ func runDisplayIndexTests(baseURL string) {
...
+	if err != nil {
+		t.FailNow()
+	}
+
+	err = test.WriteImageToFileSystem(image2, "b-repo", "tag2", storeController)
+	if err != nil {
+		t.FailNow()
+	}
+
+	err = test.WriteImageToFileSystem(image2, "b-repo", "tag2", storeController)
```
Comment (andaaron) [inline pkg/cli/image_cmd_test.go:1394]: Duplicate code.

File: pkg/cli/cve_cmd_test.go [hunk 2]
```diff
@@ -910,6 +910,92 @@ func TestServerCVEResponse(t *testing.T) {
...
+
+	cm := test.NewControllerManager(ctlr)
+	cm.StartAndWait(conf.HTTP.Port)
+
+	defer cm.StopServer()
+
+	space := regexp.MustCompile(`\s+`)
+
+	Convey("test sorting", t, func() {
+		args := []string{"list", "repo:tag", "--sort-by", "severity", "--url", baseURL}
```
Comment (andaaron) [inline pkg/cli/cve_cmd_test.go:952]: Ideally we would test the output too, should be possible if we mock.

**Review by andaaron — DISMISSED** [review]

**Comment by rchincha** [comment]
@laurentiuNiculae add examples in this PR conversation itself and cc: @mbshields

**Comment by laurentiuNiculae** [comment]
The `sort-by` flag is optional and can be used as such:

`$ zli image list --sort-by alpha-asc --config zot-config`

`$ zli image cve --sort-by severity --config zot-config`

`$ zli search query --sort-by relevance --config zot-config`

Each command has it's own list of sorting criteria:
```
list base     [ update-time, alpha-asc, alpha-dsc ]
list derived  [ update-time, alpha-asc, alpha-dsc ]
list digest   [ update-time, alpha-asc, alpha-dsc ]
list list     [ update-time, alpha-asc, alpha-dsc ]
list name     [ update-time, alpha-asc, alpha-dsc ]
list cve      [ severity   , alpha-asc, alpha-dsc ]
```
```
cve affected [ update-time, alpha-asc, alpha-dsc ]
cve fixed    [ update-time, alpha-asc, alpha-dsc ]
cve list     [ severity   , alpha-asc, alpha-dsc ]
```
```
search query   [ relevance, update-time, alpha-asc, alpha-dsc ]
search subject [ update-time, alpha-asc, alpha-dsc ]
```
```
repo list [ alpha-asc, alpha-dsc ]
```

**Pull request merged**